src/Microsoft.OData.Edm/ExtensionMethods/EdmTypeSemantics.cs

```diff
@@ -15,6 +17,9 @@ namespace Microsoft.OData.Edm
     /// </summary>
     public static class EdmTypeSemantics
     {
+
+        private static ConcurrentDictionary<string, HashSet<string>> typeDictionaryCache = new ConcurrentDictionary<string, HashSet<string>>();

```

Comment:
>typeDictionaryCache [](start = 69, length = 19)

pick a more descriptive name --- what types of types are in this dictionary? what's it used for?  maybe something like "baseTypeCache?